src/components/DayPickerKeyboardShortcuts.jsx

```diff
@@ -97,7 +70,98 @@ class DayPickerKeyboardShortcuts extends React.Component {
       label: phrases.questionMark,
       action: phrases.openThisPanel,
     },
-    ];
+  ];
+}
+
+class DayPickerKeyboardShortcuts extends React.Component {
+  constructor(...args) {
+    super(...args);
+
+    this.keyboardShortcuts = getKeyboardShortcuts(this.props.phrases);
+
+    this.onShowKeyboardShortcutsButtonClick = this.onShowKeyboardShortcutsButtonClick.bind(this);
+    this.setShowKeyboardShortcutsButtonRef = this.setShowKeyboardShortcutsButtonRef.bind(this);
+    this.setHideKeyboardShortcutsButtonRef = this.setHideKeyboardShortcutsButtonRef.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.phrases !== this.props.phrases) {
+      this.keyboardShortcuts = getKeyboardShortcuts(nextProps.phrases);
+    }
+  }
+
+  componentDidUpdate() {
+    this.handleFocus();
+  }
+
+  onKeyDown(e) {
+    const { closeKeyboardShortcutsPanel } = this.props;
+    // Because the close button is the only focusable element inside of the panel, this
+    // amount to a very basic focus trap. The user can exit the panel by "pressing" the
+    // close button or hitting escape
+    switch (e.key) {
+      case 'Space':
+      case 'Escape':
+        e.stopPropagation();
+        closeKeyboardShortcutsPanel();
+        break;
+
+      case 'Tab':
+      case 'Enter':
+      case 'Home':
+      case 'End':
+      case 'PageUp':
+      case 'PageDown':
+      case 'ArrowLeft':
+      case 'ArrowRight':
+        e.stopPropagation();
```

Comment:
Can you clarify why you're stopping propagation here and below? and preventing default?